adam/learner/perception_graph_template.py

```diff
@@ -278,7 +278,11 @@ def copy_replacing_nodes(
         """
         return PerceptionGraphTemplate(
             graph_pattern=self.graph_pattern.copy_replacing_nodes(current_to_new_node),
-            template_variable_to_pattern_node=self.template_variable_to_pattern_node,
+            template_variable_to_pattern_node=immutabledict(
+                (variable, node)
+                for variable, node in self.template_variable_to_pattern_node.items()
+                if node not in current_to_new_node
```

Comment:
So this omits replaced nodes from the template? Wouldn't we want to replace them?

Reply:
I don't think I can get away with an easy replacement as the node I'll replace too will change into a new semantic node when the object learners process and we'll be unable to perform the update to the affordance semantic node when that occurs.